packages/server/graphql/intranetSchema/mutations/backupOrganization.ts

```diff
@@ -2,6 +2,165 @@ import {GraphQLID, GraphQLList, GraphQLNonNull, GraphQLString} from 'graphql'
+import {getTeamsByOrgIdQuery} from '../../../postgres/queries/generated/getTeamsByOrgIdQuery'
+import {insertTeamsQuery} from '../../../postgres/queries/generated/insertTeamsQuery'
+import {getGitHubAuthByTeamIdQuery} from '../../../postgres/queries/generated/getGitHubAuthByTeamIdQuery'
+import {insertGitHubAuthWithAllColumnsQuery} from '../../../postgres/queries/generated/insertGitHubAuthWithAllColumnsQuery'
+import {getDiscussionByTeamIdQuery} from '../../../postgres/queries/generated/getDiscussionByTeamIdQuery'
+import {insertDiscussionWithAllColumnsQuery} from '../../../postgres/queries/generated/insertDiscussionWithAllColumnsQuery'
+import {getUsersByIdQuery} from '../../../postgres/queries/generated/getUsersByIdQuery'
+import {insertUserWithAllColumnsQuery} from '../../../postgres/queries/generated/insertUserWithAllColumnsQuery'
+import {getTemplateRefByIdsQuery} from '../../../postgres/queries/generated/getTemplateRefByIdsQuery'
+import {insertTemplateRefWithAllColumnsQuery} from '../../../postgres/queries/generated/insertTemplateRefWithAllColumnsQuery'
+import {getTemplateScaleRefByIdsQuery} from '../../../postgres/queries/generated/getTemplateScaleRefByIdsQuery'
+import {insertTemplateScaleRefWithAllColumnsQuery} from '../../../postgres/queries/generated/insertTemplateScaleRefWithAllColumnsQuery'
+import PROD from '../../../PROD'
+import runExecFilePromise from '../../../utils/runExecFilePromise'
+
+const PG_ORG_BACKUP_DB_NAME = 'orgBackup'
+
+const createEmptyOrgBackupDB = async () => {
+  const pg_scripts_dir = 'packages/server/postgres/scripts'
+  const schemaDumpFile = 'artifacts/schemaDump.tar.gz'
+
+  const dumpScriptPath = path.resolve(process.cwd(), pg_scripts_dir, 'dump.sh')
+  const createScriptPath = path.resolve(process.cwd(), pg_scripts_dir, 'createDB.sh')
+  const restoreScriptPath = path.resolve(process.cwd(), pg_scripts_dir, 'restoreDB.sh')
+
+  await runExecFilePromise(dumpScriptPath, [`-Fc --schema-only -f ${schemaDumpFile}`])
+  await runExecFilePromise(createScriptPath, [PG_ORG_BACKUP_DB_NAME])
+  await runExecFilePromise(restoreScriptPath, [`-d ${PG_ORG_BACKUP_DB_NAME} ${schemaDumpFile}`])
+}
+
+// make insert use seq generator so tables with seq will has correct seq values
+const removeId = (row) => Object.assign(row, {id: undefined})
+
+const backupPgOrganization = async (orgIds: string[]) => {
+  await createEmptyOrgBackupDB()
+
+  const mainPg = getPg()
+  const mainClient = await mainPg.connect()
+
+  const orgBackupConfig = Object.assign(getPgConfig(), {
+    database: PG_ORG_BACKUP_DB_NAME,
+    max: PROD ? 5 : 1
+  })
+  const orgBackupClient = new Client(orgBackupConfig)
+  await orgBackupClient.connect()
+
+  // rethink client for when we need to join with rethink
+  const r = await getRethink()
+
+  // collect all backup writes here
+  const writePromises: Promise<any>[] = []
+
+  try {
+    const [migrations, postMigrations, auditRows, teams] = await Promise.all([
```

Comment:
big picture, this strategy reads ALL the data from PG, puts it in memory in node, then moves ALL that data back to PG.
this is a little dangerous:
- it will cause node to run out of memory & crash
- inserts can only handle 16-bit params, so if it's more than a couple thousand rows it'll fail

i think what you want is to:
- clone the schema
- SELECT INTO from the source schema to the table in the target schema

Reply:
hey @mattkrick i did write my approach in a [comment on the original issue](https://github.com/ParabolInc/parabol/issues/5043#issuecomment-881686121). It explains how I first tried to use `CREATE TABLE AS` (functionally equivalent to `SELECT INTO`) and why I chose not to go that route.

this strategy doesn't read all the data from pg, only the data which is relevant to the org. the only exception is it does read all migration data, since we need that regardless.